src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java

```diff
@@ -499,6 +491,15 @@ private void validateActivity(final Long activityId) {
         return new PageImpl<>(visitsWithVisitors, pageable, visits.getTotalElements());
     }
 
+    private RelationshipType getRelationshipType(final VisitInformationFilter filter, final Long personId) {
+        if (personId == null) return null;
+        return offenderContactPersonsRepository.findAllByPersonIdAndOffenderBooking_BookingId(personId, filter.getBookingId())
```

Comment:
I think you could have a repository method like `findFirstBy....OrderByLastUpdated` or something like that